app/models/user.rb

```diff
@@ -0,0 +1,32 @@
+require 'data_mapper'
+require 'dm-postgres-adapter'
+require 'bcrypt'
+
+class User
+
+  attr_reader :password
+  attr_accessor :password_confirmation
+
+  include DataMapper::Resource
+  property :id, Serial
+  property :name, String, required: true
+  property :username, String, required: true, unique: true
+  property :email, String, required: true, unique: true
+  property :password_digest, Text
+
+  validates_confirmation_of :password, message: 'Password confirmation incorrect'
+  validates_format_of :email, as: :email_address, message: 'Please enter a valid email address'
```

Comment:
Metrics/LineLength: Line is too long. [95/90] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)